### DISCUSSION TITLE
Show your clean architecture implementation

### DISCUSSION THREAD

**Discussion by rajasingh012** [jasontaylordev/CleanArchitecture#631] [General]
I hope many of you created project based on this template, if you  project is open source, share the link.  What are the problems you have faced when you implemented real world scenarios and how you fix it. Share your thoughts.

Through our discussion, community can learn better

**Comment by iayti** [comment]
I added some extra stuff and change some libraries (AutoMapper to Mapster for better performance and more easy to implement). This is too early but i am updating.

Future implementations:
1) Logging into Elasticsearch using Serilog and viewing logs in Kibana
2) Adding Blazor WebUI
3) Adding Redis
4) Adding RabitMQ/Azure Service Bus (I don't decide yet 😃 )
5) Hangfire and Worker Service example for service jobs.
6) Adding microservice implementation example (Dapr/Ocelot)
7) Collect metrics and create dashboards using Grafana, Prometheus and AppMetrics in .NET Core

I intend to divide the releases here into themes from basic to advanced.

Thank you in advance for your constructive criticism.
Please Check my template.
https://github.com/iayti/CleanArchitecture

**Comment by shahabfar** [comment]
> I have a bit of understanding issue in regards to the audit properties, I didn't get why CreatedBy and UpdatedBy are string values instead of the type of the user primary key property?

Simply write Username which has string type into these two parameters.

**Comment by MrJohnRiceHimself** [comment]
I modified the ApplicationDbContext and Configurations in Infrastructure to implement soft-delete and multi-tenancy. Needs more testing before I'd consider sharing though. Work in progress, I'll PR if I ever finish... 👍

**Comment by iayti** [comment]
> I added some extra stuff and change some libraries (AutoMapper to Mapster for better performance and more easy to implement). This is too early but i am updating.
> 
> Future implementations:
> 
> 1. Logging into Elasticsearch using Serilog and viewing logs in Kibana
> 2. Adding Blazor WebUI
> 3. Adding Redis
> 4. Adding RabitMQ/Azure Service Bus (I don't decide yet 😃 )
> 5. Hangfire and Worker Service example for service jobs.
> 6. Adding microservice implementation example (Dapr/Ocelot)
> 7. Collect metrics and create dashboards using Grafana, Prometheus and AppMetrics in .NET Core
> 
> I intend to divide the releases here into themes from basic to advanced.
> 
> Thank you in advance for your constructive criticism.
> Please Check my template.
> https://github.com/iayti/CleanArchitecture

project updated to .net 5 😄

**Comment by digaomatias** [comment]
Hey @iayti, I've forked your project, so I can do my take on it. 
I prefer Mapster over Automapper as well. I think one of the changes I'll do is to replace NUnit for XUnit. 

I got a question, what is the WorkerService project, and what is it used for? Could you provide some documentation (or links to it) about it?

**Comment by iayti** [comment]
> Hey @iayti, I've forked your project, so I can do my take on it.
> I prefer Mapster over Automapper as well. I think one of the changes I'll do is to replace NUnit for XUnit.
> 
> I got a question, what is the WorkerService project, and what is it used for? Could you provide some documentation (or links to it) about it?

Hi @digaomatias 

WorkerService is not a required project, it is not connected to anything. It can be deleted directly from the project. I want to add a Blazor WebUI project in the future. I created the WorkerService project to pre-prepare for WebUI. When the WebApi project is build, the codes automatically generated with NSwag are added under the Client folder in WorkerService project. WorkerService is used for NSwag auto code generation example 😊 

Have a nice coding

**Comment by Ricard01** [comment]
I really like the clean architecture template so i made one from scratch so can i understand how it works. It took me a while and finaly made and web api that can help others to easy understand  and deploy with this architecture hope it helps.

I Plan to use it as a template for production so thank you in advance for your constructive criticism.
https://github.com/Ricard01/Knowledge.git

**Comment by neozhu** [comment]
I have developed a Blazor application project based on this clean architecture.
please see this link
https://github.com/neozhu/CleanArchitectureWithBlazorServer
